Automattic/newspack-plugin
# feat: campaign groups in popup action card

### All Submissions:






### Changes proposed in this Pull Request:

Adds UI to the Campaigns Wizard to control the Campaign Groups assigned to a Campaign. 

Dependent PRs:
~https://github.com/Automattic/newspack-plugin/pull/764~

This branch must be tested along with https://github.com/Automattic/newspack-popups/pull/363

<img width="1401" alt="Screen Shot 2020-12-30 at 6 32 09 PM" src="https://user-images.githubusercontent.com/1477002/103386973-4d453900-4acf-11eb-8d42-3a255bf08d02.png">

### How to test the changes in this Pull Request:

Switch `newspack-popups` to `feat/campaign-groups-in-popup-action-card`.

1. Make sure there are some Campaign Groups defined. This can be done by editing any campaign and adding campaign groups in the pane near the bottom of the sidebar. Note: at a later point we'll add UI to create Campaign Groups directly in the Campaigns wizard but this would be premature.
2. In the Campaigns wizard, click the hamburger icon on any Campaign. Observe a `Campaign Groups` autocomplete field.
3. Begin typing one of the campaign group names, observe autocomplete finds it. Click the name and observe it is added.
4. Refresh the page and observe the assigned campaign groups remain. 
5. Try removing a campaign group and observe this works as expected.
6. For Sitewide default Overlay Campaigns, observe that the hamburger tooltip reads `Campaign groups` instead of `Category filtering and campaign groups`. Click the hamburger and observe that only the Campaign Groups UI is visible.
7. Verify that the `Category filtering` fields work as they always have.

### Other information:
